### PR TITLE
List routes endpoint

### PR DESCRIPTION
**Pull request by JohnCleese**

### CODE REVIEW

File: backend/fleet_management/api.py [hunk 1]
```diff
@@ -1,11 +1,33 @@
-from rest_framework import views
+from django_filters.rest_framework import DjangoFilterBackend
+from rest_framework import generics, filters, views
 from rest_framework.response import Response
-
-from .serializers import UserSerializer
+from .serializers import UserSerializer, RouteSerializer
+from .models import Route
+from rest_framework.permissions import IsAuthenticated
```
Comment (jacol12345) [inline backend/fleet_management/api.py]: Please fix order of imports.

File: backend/fleet_management/api.py [hunk 2]
```diff
@@ -1,11 +1,33 @@
...
 
 class CurrentUserRetrieveView(views.APIView):
     def get(self, request):
         return Response(
             UserSerializer(request.user).data
         )
+
+
+class RouteListView(generics.ListAPIView):
+    authentication_classes = (IsAuthenticated,)
```
Comment (jacol12345) [inline backend/fleet_management/api.py]: IMO not needed.

File: backend/fleet_management/api.py [hunk 3]
```diff
@@ -1,11 +1,33 @@
...
+    serializer_class = RouteSerializer
+    filter_backends = (filters.OrderingFilter, filters.SearchFilter, DjangoFilterBackend,)
+    search_fields = (
+        'car',
```
Comment (jacol12345) [inline backend/fleet_management/api.py]: Does it work with filtering by related objects? How does it know which field to use?

File: backend/fleet_management/models.py [hunk 4]
```diff
@@ -27,9 +27,11 @@ def __str__(self):
...
     driver = models.ForeignKey(User, on_delete=models.CASCADE)
     car = models.ForeignKey(Car, null=False, on_delete=models.CASCADE)
+    project = models.ForeignKey(Project, on_delete=models.CASCADE)
     passengers = models.ManyToManyField(Passenger)
     date = models.DateField(default=now, blank=False)
     start_mileage = models.IntegerField(null=False)
     end_mileage = models.IntegerField(null=False)
-    fuel_level_begin = models.FloatField(null=False)
-    fuel_level_end = models.FloatField(null=False)
+    description = models.CharField(max_length=500, blank=True)
```
Comment (jacol12345) [inline backend/fleet_management/models.py]: Description is required (aka what was the purpose of the drive).

File: backend/fleet_management/serializers.py [hunk 5]
```diff
@@ -1,9 +1,26 @@
...
     class Meta:
         model = User
         fields = ('id', 'username')
+
+
+class RouteSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Route
+        fields = (
+            'car',
```
Comment (jacol12345) [inline backend/fleet_management/serializers.py]: Can you remind how will i.e. car be serialized?

Comment (JohnCleese) [reply backend/fleet_management/serializers.py]: by id?

Comment (jacol12345) [reply backend/fleet_management/serializers.py]: I think we will more than id to show car data. Take a look at nesting serializers: https://www.django-rest-framework.org/api-guide/relations/#example

**Comment by jacol12345** [comment]
Already resolved.

**Pull request closed without merging**